test/micro/org/openjdk/bench/javax/crypto/full/AESGCMBench.java

```diff
@@ -26,41 +26,38 @@
 import org.openjdk.jmh.annotations.Param;
 import org.openjdk.jmh.annotations.Setup;
 
-import javax.crypto.BadPaddingException;
 import javax.crypto.Cipher;
-import javax.crypto.IllegalBlockSizeException;
-import javax.crypto.NoSuchPaddingException;
 import javax.crypto.spec.GCMParameterSpec;
 import javax.crypto.spec.SecretKeySpec;
-import java.security.InvalidAlgorithmParameterException;
-import java.security.InvalidKeyException;
-import java.security.NoSuchAlgorithmException;
-import java.security.spec.InvalidParameterSpecException;
 
+/**
+ * This performance tests runs AES/GCM encryption and decryption using byte[]
+ * as input and output buffers for single and multi-part testing.
+ *
+ * This test rotates the IV and creates a new GCMParameterSpec for each encrypt
+ * benchmark operation
+ */
 
 public class AESGCMBench extends CryptoBase {
 
-    @Param({"AES/GCM/NoPadding"})
-    private String algorithm;
-
     @Param({"128"})
     private int keyLength;
 
-    @Param({""+16*1024})
+    @Param({"" + 1024, "" +  1500, "" + 4096, "" + 16384})
     private int dataSize;
 
-    byte[] data;
     byte[] encryptedData;
+    byte[] in, out;
     private Cipher encryptCipher;
     private Cipher decryptCipher;
     SecretKeySpec ks;
     GCMParameterSpec gcm_spec;
-    byte[] aad;
```

Comment:
Not an expert on this, but why no more AAD?

Reply:
For a performance test there isn't any value in adding an AAD.  It's data sitting in a buffer that an be easily replicated with a multipart operation